### PR TITLE
feat(ssr): support custom link header

### PR DESCRIPTION
**Pull request by tillkruss**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description
It's currently impossible to overwrite or extend the `Link` header sent by Nuxt.

We have a middleware that needs to add a couple of `preconnect` links:

```
<https://cdn.example.com>; rel=preconnect; crossorigin
```

## Checklist:
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

File: packages/core/src/middleware/nuxt.js [hunk 1]
```diff
@@ -64,10 +64,15 @@ export default async function nuxtMiddleware(req, res, next) {
...
+      // Append existing Link header to end of list
+      if (res.getHeader('Link')) {
+        pushAssets.push(res.getHeader('Link'))
+      }
+
       // Pass with single Link header
       // https://blog.cloudflare.com/http-2-server-push-with-multiple-assets-per-link-header
       // https://www.w3.org/Protocols/9707-link-header.html
-      res.setHeader('Link', pushAssets.join(','))
+      res.setHeader('Link', pushAssets.join(', '))
```
Comment (tillkruss) [inline packages/core/src/middleware/nuxt.js]: We're already using spaces after semicolons in line 64.

**Review by TheAlexLichter — APPROVED** [review]
LGTM!

**Comment by awronski** [comment]
Hi Guys!

What do you think about this one?
https://cmty.app/nuxt/nuxt.js/issues/c7870

**Comment by manniL** [comment]
@awronski Of course, implementing https://github.com/nuxt/nuxt.js/issues/4011 and giving the user the choice would open up more possibilities!

A PR would be appreciated :clap:

**Comment by tillkruss** [comment]
@awronski: I'd prefer a function, too.

**Review by clarkdo — COMMENTED** [review]
Good idea and if you can add test for it, it will be perfect then😊

**Review by galvez — APPROVED** [review]
LGTM

**Review by pi0 — APPROVED** [review]
LGTM. Adding a test can make it a perfect one!

**Comment by tillkruss** [comment]
@pi0, @clarkdo: I'd love to add a test, but I have to admit I wouldn't even know where to start.

**Comment by galvez** [comment]
@tillkruss I'll take care of it :)

**Comment by tillkruss** [comment]
We can close this in favour of #4198 IMO.

**Comment by manniL** [comment]
closed in favor of #4198

**Pull request closed without merging**